test/e2e/network/endpointslice.go

```diff
@@ -303,6 +304,133 @@ var _ = SIGDescribe("EndpointSlice", func() {
 		deleteEndpointSlices(cs, f.Namespace.Name, svc2)
 		expectEndpointsAndSlices(cs, f.Namespace.Name, svc2, []*v1.Pod{pod1, pod2}, 2, 2, true)
 	})
+
+	ginkgo.It("should NOT include terminating endpoints for Endpoints and EndpointSlices", func() {
+		labelPod1 := "pod1"
+		labelValue := "on"
+
+		pod1 := podClient.Create(&v1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "pod1",
+				Labels: map[string]string{
+					labelPod1: labelValue,
+				},
+			},
+			Spec: v1.PodSpec{
+				// use long grace periods to leave time to test endpoints
+				TerminationGracePeriodSeconds: utilpointer.Int64Ptr(120),
```

Comment:
Not an e2e test expert but does this risk leaving test pods/namespaces around longer than usual?

Reply:
Yeah the namespace termination would take longer. If there are multiple runners it probably wouldn't impact test time that much but serial tests would take extra time for namespace garbage collection. Not sure there's really a workaround for this aside from reducing this value, but that leaves less time to validate terminating endpoints. Maybe 60s is enough headroom.

Reply:
we have some constants for endpoints timeouts, let´s use them https://github.com/kubernetes/kubernetes/blob/b259c92dda81e18822b790501445d98d7c0ed339/test/e2e/framework/service/const.go#L33-L45

Reply:
Those are kube-proxy constants though? Will re-use the same timeout values though.

Reply:
yeah, but they are there because is an endpoint consumer, I thought that can give us a reference

Reply:
Good point, 30s seems like a reasonable baseline here. I think I'm actually going to make it stricter at 10s since this is mostly an API only test so it's unlikely to see the same level of flakiness as something like kube-proxy